pkg/serverless/invocationlifecycle/lifecycle.go

```diff
@@ -230,6 +229,23 @@ func (lp *LifecycleProcessor) OnInvokeStart(startDetails *InvocationStartDetails
 		}
 		ev = event
 		lp.initFromLambdaFunctionURLEvent(event, region, account, resource)
+	case trigger.LegacyStepFunctionEvent:
+		var event events.StepFunctionEvent
+		if err := json.Unmarshal(payloadBytes, &event); err != nil {
+			log.Debugf("Failed to unmarshal %s event: %s", stepFunction, err)
+			break
+		}
+		ev = event.Payload
+		lp.initFromStepFunctionEvent(event)
+	case trigger.StepFunctionEvent:
+		var eventPayload events.StepFunctionPayload
+		if err := json.Unmarshal(payloadBytes, &eventPayload); err != nil {
+			log.Debugf("Failed to unmarshal %s event: %s", stepFunction, err)
+			break
+		}
+		sfe := events.StepFunctionEvent{Payload: eventPayload}
```

Comment:
If you set the event to the full event and not just `event.Payload` above, you wouldn't need to do this here.

Reply:
Because Legacy and non-Legacy step function lambda invocations unmarshal into different types, we have to do a little awkward juggling regardless.

Reply:
Ahhh, I think you mentioned this before. Can you point me again to what those two json blobs look like? I'm now wondering if there's a way we could unify them into a single event type, even if they look a bit different from one another.

Reply:
Legacy:
```
{
  "Payload": {
    "Execution": {
      "Id": "arn:aws:states:us-east-1:425362996713:execution:agocsTestSF:bc9f281c-3daa-4e5a-9a60-471a3810bf44",
      "Input": {},
      "StartTime": "2024-07-30T19:55:52.976Z",
      "Name": "bc9f281c-3daa-4e5a-9a60-471a3810bf44",
      "RoleArn": "arn:aws:iam::425362996713:role/test-serverless-stepfunctions-dev-AgocsTestSFRole-tRkeFXScjyk4",
      "RedriveCount": 0
    },
    "StateMachine": {
      "Id": "arn:aws:states:us-east-1:425362996713:stateMachine:agocsTestSF",
      "Name": "agocsTestSF"
    },
    "State": {
      "Name": "agocsTest1",
      "EnteredTime": "2024-07-30T19:55:53.018Z",
      "RetryCount": 0
    }
  }
}
```

Non-legacy

```
{
  "Execution": {
    "Id": "arn:aws:states:us-east-1:425362996713:execution:agocsTestSF:bc9f281c-3daa-4e5a-9a60-471a3810bf44",
    "Input": {},
    "StartTime": "2024-07-30T19:55:52.976Z",
    "Name": "bc9f281c-3daa-4e5a-9a60-471a3810bf44",
    "RoleArn": "arn:aws:iam::425362996713:role/test-serverless-stepfunctions-dev-AgocsTestSFRole-tRkeFXScjyk4",
    "RedriveCount": 0
  },
  "StateMachine": {
    "Id": "arn:aws:states:us-east-1:425362996713:stateMachine:agocsTestSF",
    "Name": "agocsTestSF"
  },
  "State": {
    "Name": "agocsTest1",
    "EnteredTime": "2024-07-30T19:55:53.018Z",
    "RetryCount": 0
  }
}
```